docs/source-2.0/guides/model-validation-examples.rst

```diff
@@ -0,0 +1,619 @@
+            """,
+            selector: ":not([@id|name: @{(length)} <= 60 && @{(length)} >= 3])"
+        }
+    }
+
+
+Limit Nesting Depth of Input and Output shape members
+==============================================================
+
+This example checks that resources are not deeply nested. In this case, it will check that the nesting
+depth is less than 4.
+
+.. code-block:: smithy
+
+    {
+        "name": "EmitEachSelector",
+        "id": "LimitNestingDepthToFourLayers",
+        "configuration": {
+            messageTemplate: """
+                `@{id}`, bound to operation `@{var|operation}` has a nesting depth >4. This is typically not advised.
+                You should look for ways to reduce the nesting depth of this shape.
+                """,
+            selector: """
+                $operation(*) -[input, output, error]->
+                :test(> member > * > member > * > member > * member > *)
+                """
+        }
+    }
+
+
+Operations should have documentation
+============================================
+
+This example checks for documentation on all operation shapes.
+
+.. code-block:: smithy
+
+    {
+        name: "EmitEachSelector",
+        id: "MissingOperationDocumentation",
+        configuration: {
+            messageTemplate: """
+                Operation `@{id|name}` is missing documentation. Add the `@@documentation` \
+                trait to this operation.
+                """,
+            selector: "operation :not([trait|documentation])"
+        }
+    }
+
+
+Examples on all operations
+============================================
+
+This example checks for examples on all operation shapes.
+
+.. code-block:: smithy
+
+    {
+        name: "EmitEachSelector",
+        id: "MissingOperationExamples",
+        configuration: {
+            messageTemplate: """
+                Operation `@{id|name}` is missing examples. Add the `@@examples` \
+                trait to this operation.
+                """,
+            selector: "operation :not([trait|examples])"
+        }
+    }
+
+
+Operations should have common exception
+============================================
+
+The following example shows how to check that all operations throw a common exception.
+
+.. code-block:: smithy
+
+    {
+        name: "EmitEachSelector",
+        id: "OperationErrorsIncludesCommonException",
+        configuration: {
+            messageTemplate: """
+            Expected error `CommonException` is not bound to operation @{id|name} . Add \
+            `CommonException` to the list of errors bound to this operation.
+            """
+            selector: "operation :not(:test(-[error]-> [id|name=CommonException]))",
+        }
+    }
+
+
+.. note::
+    If you want an operation to throw multiple common errors you likely want to use a validator that
+    checks for a common mixin rather than a single operation. See :ref:`Operations Should Use Common Mixin <operations-should-use-common-mixin>`
```

Comment:
This is more of a `tip::`, since it's a helpful suggestion to the reader, also "errors you likely" should be "errors, you likely"